assets/wizards/advertising/views/ad-units/index.js

```diff
@@ -5,97 +5,168 @@
 /**
  * WordPress dependencies
  */
+import { useState, Fragment } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
-
-/**
- * External dependencies
- */
-import classnames from 'classnames';
+import { trash, pencil } from '@wordpress/icons';
 
 /**
  * Internal dependencies
  */
-import { ActionCard, Notice, withWizardScreen } from '../../../../components/src';
+import {
+	ActionCard,
+	TextControl,
+	Button,
+	Card,
+	Notice,
+	withWizardScreen,
+} from '../../../../components/src';
 
 /**
  * Advertising management screen.
  */
-class AdUnits extends Component {
-	/**
-	 * Render.
-	 */
-	render() {
-		const { adUnits, onDelete, service, isGAMConnected } = this.props;
+const AdUnits = ( {
+	adUnits,
+	onDelete,
+	updateAdUnit,
+	wizardApiFetch,
+	service,
+	gamConnectionStatus = {},
+} ) => {
+	const warningNoticeText = `${ __(
+		'Please connect your Google account using the Newspack dashboard in order to use ad units from your GAM account.',
+		'newspack'
+	) } ${
+		Object.values( adUnits ).length
+			? __( 'The legacy ad units will continue to work, but they are not editable.', 'newspack' )
+			: ''
+	}`;
+	const gamConnectionMessage = gamConnectionStatus?.error
+		? `${ __( 'Google Ad Manager connection error', 'newspack' ) }: ${ gamConnectionStatus.error }`
+		: false;
 
-		return (
-			<Fragment>
-				{ isGAMConnected === false && (
-					<Notice
-						noticeText={ `${ __(
-							'Please connect your Google account using the Newspack dashboard in order to use ad units from your GAM account.',
-							'newspack'
-						) } ${
-							Object.values( adUnits ).length
-								? __(
-										'The legacy ad units will continue to work, but they are not editable.',
-										'newspack'
-								  )
-								: ''
-						}` }
-						isWarning
+	const isDisplayingNetworkMismatchNotice =
+		! gamConnectionMessage && false === gamConnectionStatus?.is_network_code_matched;
+
+	const [ networkCode, setNetworkCode ] = useState( gamConnectionStatus.network_code );
+	const saveNetworkCode = () => {
+		wizardApiFetch( {
+			path: '/newspack/v1/wizard/advertising/network_code/',
+			method: 'POST',
+			data: { network_code: networkCode },
+			quiet: true,
+		} );
+	};
+
+	return (
+		<Fragment>
```

Comment:
Is this extra wrapping `Fragment` necessary? Can't we return before the final render if `gamConnectionStatus.can_connect` is false? At the very least maybe we can stick with `<>` since we're not applying any props.

Reply:
> Is this extra wrapping Fragment necessary? 

That's a list of children components 🤷‍♀️

> Can't we return before the final render if gamConnectionStatus.can_connect is false?

The legacy ad units have to remain editable, even if GAM is unavailable.

> At the very least maybe we can stick with <> since we're not applying any props.

Done in bb65fde8276a12fd3e56c6c98c1a0cf5380c9ff0

Reply:
Oh, I see what you mean. Thanks for explaining!